ogr/ogrsf_frmts/adbc/ogradbcdataset.cpp

```diff
@@ -0,0 +1,512 @@
+            return false;
+        }
+    }
+    else if (pszFilename[0])
+    {
+        if (AdbcDatabaseSetOption(&m_database, "uri", pszFilename, error) !=
+            ADBC_STATUS_OK)
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "AdbcDatabaseSetOption() failed: %s", error.message());
+            return false;
+        }
+    }
+
+    for (const auto &[pszKey, pszValue] : cpl::IterateNameValue(
+             static_cast<CSLConstList>(poOpenInfo->papszOpenOptions)))
+    {
+        if (STARTS_WITH_CI(pszKey, "ADBC_OPTION_"))
+        {
+            if (AdbcDatabaseSetOption(&m_database,
+                                      pszKey + strlen("ADBC_OPTION_"), pszValue,
+                                      error) != ADBC_STATUS_OK)
+            {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "AdbcDatabaseSetOption() failed: %s", error.message());
+                return false;
+            }
+        }
+    }
+
+    if (AdbcDatabaseInit(&m_database, error) != ADBC_STATUS_OK)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "AdbcDatabaseInit() failed: %s",
+                 error.message());
+        return false;
+    }
+
+    m_connection = std::make_unique<AdbcConnection>();
+    if (AdbcConnectionNew(m_connection.get(), error) != ADBC_STATUS_OK)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "AdbcConnectionNew() failed: %s",
+                 error.message());
+        return false;
+    }
+
+    if (AdbcConnectionInit(m_connection.get(), &m_database, error) !=
+        ADBC_STATUS_OK)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "AdbcConnectionInit() failed: %s",
+                 error.message());
+        return false;
+    }
+
+    std::string osLayerName = "RESULTSET";
+    std::string osSQL;
+    const char *pszSQL = CSLFetchNameValue(poOpenInfo->papszOpenOptions, "SQL");
+    bool bIsParquetLayer = false;
+    if (bIsParquet)
+    {
+        m_osParquetFilename = pszFilename;
+        osLayerName = CPLGetBasename(pszFilename);
+        if (osLayerName == "*")
+            osLayerName = CPLGetBasename(CPLGetDirname(pszFilename));
+        if (!pszSQL)
+        {
+            osSQL = CPLSPrintf(
+                "SELECT * FROM '%s'",
+                CPLString(pszFilename).replaceAll("'", "''").c_str());
+            pszSQL = osSQL.c_str();
+            bIsParquetLayer = true;
+        }
+    }
+
+    if (pszSQL)
+    {
+        if (pszSQL[0])
+        {
+            auto poLayer = CreateLayer(pszSQL, osLayerName.c_str());
+            if (!poLayer)
+                return false;
+            poLayer->m_bIsParquetLayer = bIsParquetLayer;
+            m_apoLayers.emplace_back(std::move(poLayer));
+        }
+    }
+    else if (bIsDuckDB || bIsSQLite3)
+    {
+        auto poLayerList = CreateLayer(
+            "SELECT name FROM sqlite_master WHERE type IN ('table', 'view')",
```

Comment:
this branch is also executed when `bIsDuckDB` is it correct to query `sqlite_master` in that case?

Reply:
yes, this is fine. DuckDB immitates SQLite: https://duckdb.org/docs/guides/meta/list_tables.html#see-also